### PR TITLE
fix: Adjust autocomplete results ranking for function arguments

### LINKED ISSUE #28107: [Task] Autocomplete results should not include strings for a function parameter that is of type object

### PR DESCRIPTION
**Pull request by arunvjn**
#### PR fixes following issue(s)
Fixes #28107
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
#### How Has This Been Tested?
- [x] Manual
- [x] Jest
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/utils/autocomplete/CodemirrorTernService.ts [hunk 1]
```diff
@@ -1161,7 +1149,22 @@ class CodeMirrorTernService {
...
+    const inner = CodeMirror.innerMode(cm.getMode(), state);
+    if (inner.mode.name != "javascript") return false;
+    const lex = inner.state.lexical;
+    if (lex.info === "call") {
+      const argPos = lex.pos || 0;
+      const args = this.cachedArgHints?.type?.args || [];
+      const arg = args[argPos];
+      const argType = arg?.type;
+      entityInformation.expectedType = getDataType(argType);
+    }
```
Comment (arunvjn) [inline app/client/src/utils/autocomplete/CodemirrorTernService.ts:1167]: This is the core change

**Review by ohansFavour — APPROVED** [review]

**Pull request merged**